crates/core/src/otel.rs

```diff
@@ -23,6 +30,86 @@ pub struct OtelConfig {
     pub metrics_endpoint: Option<String>,
     /// Overrides the OpenTelemetry endpoint for logs.
     pub logs_endpoint: Option<String>,
+    /// Determines whether grpc or http will be used for exporting the telemetry.
+    pub protocol: Option<String>,
+}
+
+impl OtelConfig {
+    pub fn parse(self) -> anyhow::Result<Self> {
+        match self.protocol.clone().unwrap_or_default().as_ref() {
+            "http" | "grpc" => {}
+            protocol => bail!("unsupported protocol: {:?}", protocol),
+        };
+        Ok(self)
+    }
```

Comment:
I find this function a little confusing since it reads more like a `validate()` since it acts on existing "valid" variables, but I think that changing the stored variable to an `OtelProtocol` would remove the need for this

Reply:
Yeah this really confused me as well. But I also agree with @brooksmtownsend's suggested change

Reply:
You're exactly right, I already had `Default` impl in place for OtelProtocol, all I needed was to add `FromStr` and this could be handled with the combination of clap arguments and `unwrap_or_default()` on the arguments as they get passed in to the `OtelConfig` struct, thanks!